include/ginkgo/core/matrix/dense.hpp

```diff
@@ -122,6 +132,7 @@ class Dense
     friend class SparsityCsr<ValueType, int64>;
     friend class Dense<to_complex<ValueType>>;
     friend class experimental::distributed::Vector<ValueType>;
+    friend class gko::detail::VectorCache<ValueType>;
```

Comment:
why is this necessary?

Reply:
Because `distributed::Vector` holds the `Dense` object not `Dense` pointer, I need the constructor to construct the object directly and then use the move_to

Reply:
ah, I might use the `unique_ptr.release` from create function

Reply:
No it's fine, I just got confused. I thought this was in the `distributed::Matrix`.